### PR TITLE
QPU Parity

### PR DESCRIPTION
**Pull request by msohaibalam**
Addressing issue #20 and increasing parity between the QVM and QPU abstractions. Circuits run on e.g. wire 0 will now run on the (0-indexed) first qubit on the physical lattice.

### CODE REVIEW

**Comment by msohaibalam** [comment]
@josh146 I was initially planning to give the user an option to "reverse wire" as well. But on 2nd thought, since it seems that Pennylane's default and intended use is to use the 0..n_qubits-1 indexing convention, it might make more sense to just force the "wiring" introduced in this PR, and not allow the user to define their own. Open to suggestions/thoughts on this however.

File: pennylane_forest/qpu.py [hunk 1]
```diff
@@ -111,21 +112,27 @@ def pre_rotations(self, observable, wires):
...
-            "Hadamard": [ExperimentSetting(TensorProductState(), float(np.sqrt(1/2)) * sX(0)),
-                         ExperimentSetting(TensorProductState(), float(np.sqrt(1/2)) * sZ(0))]
-        }
-        # expectation values for single-qubit observables
+        # Single-qubit observable
         if len(wires) == 1:
-
+            # identify Experiment Settings for each of the possible single-qubit observables
+            wire = wires[0]
+            qubit = self.wiring[wire]
```
Comment (josh146) [inline pennylane_forest/qpu.py]: Ends up being a very simple fix! 🙂

**Comment by msohaibalam** [comment]
> So for instance,
> 
> ```python
> dev = qml.device("forest.qpu", backend="Aspen-4", wire_numbering="physical")
> ```
> 
> which allows the user to use the physical qubit ordering?

Yes, that's exactly what I had in mind. Although I now realize [this check](https://github.com/XanaduAI/pennylane/blob/master/pennylane/qnode.py#L675) in `pennylane.qnode` will block us from doing this, and will raise an error on trying to run a circuit on e.g. wire (qubit) 10 on "Aspen-4-4Q-E". So maybe table this for now?

> I would like to, when we have time, refactor this plugin to better match PyQuil terminology and 
> UX. I.e., merge forest.qvm with forest.qpu, maybe change the device="Aspen-*" keyword 
> argument to qc="Aspen-*.

I agree, we should work this out at some point. For now, I've created an issue to address it: https://github.com/rigetti/pennylane-forest/issues/28.

**Comment by msohaibalam** [comment]
@josh146 As we discussed offline, let's first work on updating PennyLane to handle the "physical" wiring of the QPUs, and then produce that capability within PennyLane-Forest. Merging these changes into master.

**Pull request merged**